main.c

```diff
@@ -164,6 +164,9 @@ int blanked = 0;
 int must_disable_dpms = 0;
 #endif
 
+//! If set by signal handler we should wake up and prompt for auth.
+static int signal_wakeup = 0;
```

Comment:
Please change to sig_atomic_t for signal safety.

Reply:
Actually, `static volatile sig_atomic_t signal_wakeup = 0;`

Otherwise the compiler may optimize away the reading of the variable as it sees no calls writing to it. That would be bad.